daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -695,10 +695,13 @@ object Main extends Logging {
     // of compilation, where it asks for the parser)
     //
     val schemaSource = URISchemaSource(schema)
-    val pf = Timer.getResult("compiling", {
+    val res = Timer.getResult("compiling", {
       val processorFactory = compiler.compileSource(schemaSource)
       if (!processorFactory.isError) {
         val processor = processorFactory.onPath(path.getOrElse("/")).withValidationMode(mode)
+        if (processor.isError) {
+          None
+        }
```

Comment:
This doesn't actually return None from the function. It just means the result of this if statement is None, but that isn't used and so is essentially a no-op. This instead wants to be something like this:
```scala
val processor = ...
displayDiagnostics(processor)
if (processor.isError) {
  None
} else {
  Some(processor)
}
```